azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java

```diff
@@ -51,13 +54,23 @@ public AzureADGraphClient(ClientCredential clientCredential, AADAuthenticationPr
     }
 
     private String getUserMembershipsV1(String accessToken) throws IOException {
+        final boolean microsoftGraphApiBool = serviceEndpoints.isAadMicrosoftGraphApiBool();
```

Comment:
I think there is no need to use an extra flag to judge if the current service uses v1 or v2 graphAPIs. Just use the environment. For example:

    private final String V2_VERSION_ENV_FLAG = "microsoftgraph";
    private boolean aadMicrosoftGraphApiBool;
    

    public AzureADGraphClient(ClientCredential clientCredential, AADAuthenticationProperties aadAuthProps,
                              ServiceEndpointsProperties serviceEndpointsProps) {
        this.clientId = clientCredential.getClientId();
        this.clientSecret = clientCredential.getClientSecret();
        this.aadAuthenticationProperties = aadAuthProps;
        this.serviceEndpoints = serviceEndpointsProps.getServiceEndpoints(aadAuthProps.getEnvironment());

        this.initAADMicrosoftGraphApiBool(aadAuthProps.getEnvironment()));
    }

    private void initAADMicrosoftGraphApiBool(String endpointEnv) {
        this.aadMicrosoftGraphApiBool = false;
        if (endpointEnv.contains(V2_VERSION_ENV_FLAG)) {
            this.aadMicrosoftGraphApiBool = true;
        }
    }